In [2]:
import pandas as pd
co_df = pd.read_csv("Coffe_sales.csv", parse_dates = ['Date'])
co_df

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time
0,10,card,38.70,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000
1,12,card,38.70,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000
2,12,card,38.70,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000
3,13,card,28.90,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000
4,13,card,38.70,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000
...,...,...,...,...,...,...,...,...,...,...,...
3542,10,card,35.76,Cappuccino,Morning,Sun,Mar,7,3,2025-03-23,10:34:54.894000
3543,14,card,35.76,Cocoa,Afternoon,Sun,Mar,7,3,2025-03-23,14:43:37.362000
3544,14,card,35.76,Cocoa,Afternoon,Sun,Mar,7,3,2025-03-23,14:44:16.864000
3545,15,card,25.96,Americano,Afternoon,Sun,Mar,7,3,2025-03-23,15:47:28.723000


In [3]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000


In [4]:
co_df.tail()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time
3542,10,card,35.76,Cappuccino,Morning,Sun,Mar,7,3,2025-03-23,10:34:54.894000
3543,14,card,35.76,Cocoa,Afternoon,Sun,Mar,7,3,2025-03-23,14:43:37.362000
3544,14,card,35.76,Cocoa,Afternoon,Sun,Mar,7,3,2025-03-23,14:44:16.864000
3545,15,card,25.96,Americano,Afternoon,Sun,Mar,7,3,2025-03-23,15:47:28.723000
3546,18,card,35.76,Latte,Night,Sun,Mar,7,3,2025-03-23,18:11:38.635000


In [5]:
co_df.shape

(3547, 11)

In [6]:
co_df.columns

Index(['hour_of_day', 'cash_type', 'money', 'coffee_name', 'Time_of_Day',
       'Weekday', 'Month_name', 'Weekdaysort', 'Monthsort', 'Date', 'Time'],
      dtype='object')

In [7]:
co_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3547 entries, 0 to 3546
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   hour_of_day  3547 non-null   int64         
 1   cash_type    3547 non-null   object        
 2   money        3547 non-null   float64       
 3   coffee_name  3547 non-null   object        
 4   Time_of_Day  3547 non-null   object        
 5   Weekday      3547 non-null   object        
 6   Month_name   3547 non-null   object        
 7   Weekdaysort  3547 non-null   int64         
 8   Monthsort    3547 non-null   int64         
 9   Date         3547 non-null   datetime64[ns]
 10  Time         3547 non-null   object        
dtypes: datetime64[ns](1), float64(1), int64(3), object(6)
memory usage: 304.9+ KB


In [8]:
co_df.describe()

,hour_of_day,money,Weekdaysort,Monthsort,Date
count,3547.000000,3547.000000,3547.000000,3547.000000,3547
mean,14.185791,31.645216,3.845785,6.453905,2024-10-04 17:34:43.676346368
min,6.000000,18.120000,1.000000,1.000000,2024-03-01 00:00:00
25%,10.000000,27.920000,2.000000,3.000000,2024-07-17 12:00:00
50%,14.000000,32.820000,4.000000,7.000000,2024-10-10 00:00:00
75%,18.000000,35.760000,6.000000,10.000000,2025-01-11 00:00:00
max,22.000000,38.700000,7.000000,12.000000,2025-03-23 00:00:00
std,4.234010,4.877754,1.971501,3.500754,NaN


In [9]:
co_df.duplicated().sum()

0

In [10]:
co_df.isna().sum()

hour_of_day    0
cash_type      0
money          0
coffee_name    0
Time_of_Day    0
Weekday        0
Month_name     0
Weekdaysort    0
Monthsort      0
Date           0
Time           0
dtype: int64

In [11]:
Q3 = 35.760000
IQR = 35.760000 - 27.920000
up = Q3 + 1.5 * IQR
co_df[co_df['money'] > up] # -------> no outliers

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time


In [12]:
co_df['coffee_name'].value_counts()

coffee_name
Americano with Milk    809
Latte                  757
Americano              564
Cappuccino             486
Cortado                287
Hot Chocolate          276
Cocoa                  239
Espresso               129
Name: count, dtype: int64

In [48]:
# Highest Profit
co_df.groupby('coffee_name')['money'].sum().sort_values(ascending = False)

coffee_name
Latte                  26875.30
Americano with Milk    24751.12
Cappuccino             17439.14
Americano              14650.26
Hot Chocolate           9933.46
Cocoa                   8521.16
Cortado                 7384.86
Espresso                2690.28
Name: money, dtype: float64

In [49]:
# Highest Price
co_df.groupby('coffee_name')['money'].mean().sort_values(ascending = False)

coffee_name
Hot Chocolate          35.990797
Cappuccino             35.883004
Cocoa                  35.653389
Latte                  35.502378
Americano with Milk    30.594710
Americano              25.975638
Cortado                25.731220
Espresso               20.854884
Name: money, dtype: float64

In [42]:
co_df[['coffee_name','money']]

,coffee_name,money
0,Latte,38.70
1,Hot Chocolate,38.70
2,Hot Chocolate,38.70
3,Americano,28.90
4,Latte,38.70
...,...,...
3542,Cappuccino,35.76
3543,Cocoa,35.76
3544,Cocoa,35.76
3545,Americano,25.96


In [15]:
# add columns like sales_count, total_revenue and avg_price
co_df['Sales_Count'] = co_df['coffee_name'].map(co_df['coffee_name'].value_counts())
co_df['Total_Revenue'] = co_df['coffee_name'].map(co_df.groupby('coffee_name')['money'].sum())
co_df['Avg_Price'] = co_df['coffee_name'].map(co_df.groupby('coffee_name')['money'].mean())

In [16]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378


In [44]:
co_df[['Time_of_Day','Weekday','Month_name']]

,Time_of_Day,Weekday,Month_name
0,Morning,Fri,Mar
1,Afternoon,Fri,Mar
2,Afternoon,Fri,Mar
3,Afternoon,Fri,Mar
4,Afternoon,Fri,Mar
...,...,...,...
3542,Morning,Sun,Mar
3543,Afternoon,Sun,Mar
3544,Afternoon,Sun,Mar
3545,Afternoon,Sun,Mar


In [17]:
co_df['Time_of_Day'].value_counts().sort_values(ascending = False) # ------> Afternoon has the highest number of orders

Time_of_Day
Afternoon    1205
Morning      1181
Night        1161
Name: count, dtype: int64

In [18]:
co_df['Weekday'].value_counts() # -----> Tuesday has the highest number of orders

Weekday
Tue    572
Mon    544
Fri    532
Thu    510
Wed    500
Sat    470
Sun    419
Name: count, dtype: int64

In [19]:
# Knowing the percentage of the number of orders by day
co_df['Weekday'].value_counts(normalize=True) * 100  

Weekday
Tue    16.126304
Mon    15.336904
Fri    14.998590
Thu    14.378348
Wed    14.096420
Sat    13.250634
Sun    11.812800
Name: proportion, dtype: float64

In [20]:
co_df['Month_name'].value_counts() # ----> March has the highest number of orders

Month_name
Mar    494
Oct    426
Feb    423
Sep    344
Aug    272
Nov    259
Dec    259
May    241
Jul    237
Jun    223
Jan    201
Apr    168
Name: count, dtype: int64

In [21]:
# clarify the product, the period during which it is sold the most, and how much of it is sold.
co_df.groupby(['coffee_name', 'Time_of_Day']).size().reset_index(name='Sales_Count').sort_values(['coffee_name', 'Sales_Count'], ascending=[True, False]).drop_duplicates('coffee_name',  keep='first')

,coffee_name,Time_of_Day,Sales_Count
0,Americano,Afternoon,233
4,Americano with Milk,Morning,331
8,Cappuccino,Night,200
11,Cocoa,Night,106
13,Cortado,Morning,143
15,Espresso,Afternoon,56
20,Hot Chocolate,Night,147
23,Latte,Night,272


In [22]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378


In [23]:
# Knowing the period of the day with the highest sales
co_df.groupby('Time_of_Day')['money'].sum()

Time_of_Day
Afternoon    38130.04
Morning      35929.20
Night        38186.34
Name: money, dtype: float64

In [24]:
# clarify which has higher sales (weekdays or weekend)
sales_by_day = co_df.groupby('Weekday')['money'].sum()

weekday_sales = sales_by_day.loc[['Mon', 'Tue', 'Wed', 'Thu', 'Sun']].sum()
weekend_sales = sales_by_day.loc[['Sat', 'Fri']].sum()

print(f"The Weekday Sales: {weekday_sales}")
print(f"The Weekend Sales: {weekend_sales}")  # -----> so sales are highest in weekday

The Weekday Sales: 80709.4
The Weekend Sales: 31536.18


In [25]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378


In [26]:
# add Is_Expensive column
co_df['Is_Expensive'] = co_df['money'] > 38

In [27]:
# true --> expensive , false --> not
co_df['Is_Expensive'].value_counts()

Is_Expensive
False    3419
True      128
Name: count, dtype: int64

In [29]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price,Is_Expensive
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378,True
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797,True
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797,True
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638,False
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378,True


In [39]:
# month_name column --> Knowing which month has the highest sales
co_df.groupby('Month_name')['money'].sum().sort_values(ascending=False)

Month_name
Mar    15891.64
Oct    13891.16
Feb    13215.48
Sep     9988.64
Nov     8590.54
Dec     8237.74
May     8164.42
Jun     7617.76
Aug     7613.84
Jul     6915.94
Jan     6398.86
Apr     5719.56
Name: money, dtype: float64

In [50]:
# Knowing info about specific month
co_df[co_df['Month_name'] == 'Apr']

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price,Is_Expensive
175,10,card,28.90,Americano,Morning,Mon,Apr,1,4,2024-04-01,10:28:47.601000,564,14650.26,25.975638,False
176,10,card,33.80,Americano with Milk,Morning,Mon,Apr,1,4,2024-04-01,10:29:42.318000,809,24751.12,30.594710,False
177,10,card,33.80,Americano with Milk,Morning,Mon,Apr,1,4,2024-04-01,10:30:43.735000,809,24751.12,30.594710,False
178,11,card,33.80,Americano with Milk,Morning,Mon,Apr,1,4,2024-04-01,11:15:43.836000,809,24751.12,30.594710,False
179,18,card,33.80,Americano with Milk,Night,Mon,Apr,1,4,2024-04-01,18:18:42.372000,809,24751.12,30.594710,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
338,13,card,27.92,Americano,Afternoon,Tue,Apr,2,4,2024-04-30,13:41:52.325000,564,14650.26,25.975638,False
339,13,card,32.82,Americano with Milk,Afternoon,Tue,Apr,2,4,2024-04-30,13:42:59.977000,809,24751.12,30.594710,False
340,19,card,37.72,Cappuccino,Night,Tue,Apr,2,4,2024-04-30,19:19:18.139000,486,17439.14,35.883004,False
341,19,card,32.82,Americano with Milk,Night,Tue,Apr,2,4,2024-04-30,19:30:04.003000,809,24751.12,30.594710,False


In [31]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price,Is_Expensive
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378,True
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797,True
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797,True
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638,False
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378,True


In [45]:
co_df[['coffee_name','Date', 'Time']]

,coffee_name,Date,Time
0,Latte,2024-03-01,10:15:50.520000
1,Hot Chocolate,2024-03-01,12:19:22.539000
2,Hot Chocolate,2024-03-01,12:20:18.089000
3,Americano,2024-03-01,13:46:33.006000
4,Latte,2024-03-01,13:48:14.626000
...,...,...,...
3542,Cappuccino,2025-03-23,10:34:54.894000
3543,Cocoa,2025-03-23,14:43:37.362000
3544,Cocoa,2025-03-23,14:44:16.864000
3545,Americano,2025-03-23,15:47:28.723000


In [32]:
# Date column --> Shows the earliest date in the sales data
co_df['Date'].min()

Timestamp('2024-03-01 00:00:00')

In [33]:
# Shows the latest date in the sales data
co_df['Date'].max()

Timestamp('2025-03-23 00:00:00')

In [46]:
co_df[['Date', 'money']]

,Date,money
0,2024-03-01,38.70
1,2024-03-01,38.70
2,2024-03-01,38.70
3,2024-03-01,28.90
4,2024-03-01,38.70
...,...,...
3542,2025-03-23,35.76
3543,2025-03-23,35.76
3544,2025-03-23,35.76
3545,2025-03-23,25.96


In [51]:
# sales 2024, 2025
sal_24 = co_df[co_df['Date'] == '2024-03-01']['money'].sum()
sal_25 = co_df[co_df['Date'] == '2025-03-01']['money'].sum()

print(f"Sales start from march 2024 are: {sal_24}$")
print(f"Sales on march 2025 are: {sal_25}$")   # --------> So Sales decreased by about 39% compared to the same day in 2024

Sales start from march 2024 are: 396.3$
Sales on march 2025 are: 241.98$


In [35]:
# clarify the date with the highest total sales and its value
best_day = co_df.groupby(['Date'])['money'].sum().idxmax()
best_value = co_df.groupby('Date')['money'].sum().max()

print(f"Date: {best_day}, Its sales: {best_value}$")

Date: 2024-10-11 00:00:00, Its sales: 836.66$


In [47]:
co_df.head()

,hour_of_day,cash_type,money,coffee_name,Time_of_Day,Weekday,Month_name,Weekdaysort,Monthsort,Date,Time,Sales_Count,Total_Revenue,Avg_Price,Is_Expensive
0,10,card,38.7,Latte,Morning,Fri,Mar,5,3,2024-03-01,10:15:50.520000,757,26875.30,35.502378,True
1,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:19:22.539000,276,9933.46,35.990797,True
2,12,card,38.7,Hot Chocolate,Afternoon,Fri,Mar,5,3,2024-03-01,12:20:18.089000,276,9933.46,35.990797,True
3,13,card,28.9,Americano,Afternoon,Fri,Mar,5,3,2024-03-01,13:46:33.006000,564,14650.26,25.975638,False
4,13,card,38.7,Latte,Afternoon,Fri,Mar,5,3,2024-03-01,13:48:14.626000,757,26875.30,35.502378,True


Dataset Summary

Dataset Overview:
 . Contains 3547 rows with 11 features (coffee_name, money, cash_type,...)
 . Covers daily sales from early 2024 through 2025
 . Used to analyze product performance and time-based sales behavior

Product Insights:
 . The most sold coffee type is Latte
 . The most expensive product on average is Hot Chocolate
 . Price variation shows that higher price doesn’t always mean higher sales
 . Some products sell almost daily, while others are ordered less frequently

Time & Day Insights:
 . Afternoon is the peak period in both sales volume and revenue
 . Tuesday records the highest total sales among all weekdays
 . Sales on regular weekdays are higher than on the weekend
 . Sales are strongest between late morning and early afternoon hours

Monthly Insights:
 . March has the highest total sales in the dataset
 . A slight drop in sales is observed in the months following March
 . Comparing March 1, 2024 (396.3) vs March 1, 2025 (241.98) shows around a 39% decrease in revenue

Financial Insights:
 . Average transaction values remain relatively consistent despite price differences
 . No outlier or abnormal prices were detected
 . Afternoon contributes the most to total daily revenue
